services/src/main/java/org/keycloak/services/resources/admin/RealmAdminResource.java

```diff
@@ -416,10 +421,25 @@ public void updateRealmEventsConfig(final RealmEventsConfigRepresentation rep) {
         }
         
         if(dateFrom != null) {
-            query.fromDate(dateFrom);
+            SimpleDateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+            Date from = null;
+            try {
+                from = df.parse(dateFrom);
+            } catch (ParseException e) {
+                throw new BadRequestException("An unexpected server error has occurred");
```

Comment:
The error message should include what's going on. For example "Invalid value for 'dateFrom', excepted format is yyyy-MM-dd". It may no be displayed like that in admin console atm, but that's a separate issue.